### ISSUE TITLE
Update pre-compute for simple error of delta_sigma

### ISSUE THREAD

**Issue by dr-guangtou** [alexieleauthaud/DeltaSigmaPipelineHSC#47]
Now we want to calculate the error for delta_sigma using just the shape noise error.  

Based on Hironao Miyatake's code `gglens_dsigma_pz_hsc.py`: 

``` python
R = 1. - 0.365**2

gt = 0.5 / R * sum["wetsigma"] / sum["w"] / 10**12
gt_err = 0.5 / R * np.sqrt(sum["wetsigma_sq"]) / sum["w"] / 10**12
gx = 0.5 / R * sum["wexsigma"] / sum["w"] / 10**12
gx_err = 0.5 / R * np.sqrt(sum["wexsigma_sq"]) / sum["w"] / 10**12
```
where `wetsigma_sq` is: 

``` python 
sum["e_sq"][i_r] += np.sum(et**2 + ex**2)
wt = l["w"] * source_bg_inbin["w"] * Sigma_cr**-2
sum["w"][i_r] += np.sum(wt)
wetsigma = wt * Sigma_cr * et
sum["wetsigma"][i_r] += np.sum(wetsigma)
sum["wetsigma_sq"][i_r] += np.sum(wetsigma**2)
wexsigma = wt * Sigma_cr * ex
sum["wexsigma"][i_r] += np.sum(wexsigma)
sum["wexsigma_sq"][i_r] += np.sum(wexsigma**2)
```

Based on these equations, it seems that we need to add new information to the pre-compte results to correctly calulate error for delta_sigma.  Right now, in `lib/precompute_ds.py`:

```
r['sum_num_t'] = (np.sum(pprib['e_t'] * pprib['e_weight'] *
                                 pprib['inverse_sigma_crit']))
r['sum_num_x'] = (np.sum(pprib['e_x'] * pprib['e_weight'] *
                                 pprib['inverse_sigma_crit']))
r['sum_den'] = np.sum(pprib['ds_weight']
```

And we need: 

```
        r['sum_num_t_sq'] = (np.sum((pprib['e_t'] * pprib['e_weight'] *
                                     pprib['inverse_sigma_crit']) ** 2))
        r['sum_num_x_sq'] = (np.sum((pprib['e_x'] * pprib['e_weight'] *
                                     pprib['inverse_sigma_crit']) ** 2))
```
@alexieleauthaud Please help me check this.

**Comment by alexieleauthaud** [comment]
So there are two ways to compute the shape noise errors. The one in Hironao's code is the same as in Melanie's code.

The other way that I have used in the past simply assumes that the weights that are applied are inverse variance weights. What we are looking to do is to estimate the variance on the weighted mean. 
https://en.wikipedia.org/wiki/Weighted_arithmetic_mean
When the weights are all inverse variance, the error on the mean is simply sigma = sqrt(1/sum(weights))

Now that I am writing this, I realize I don't know if the formula above applied when also using a lens weight, w_lens. That needs to be checked.

**Comment by dr-guangtou** [comment]
@alexieleauthaud I see!  Thanks, so that means we actually have Melanie's version of error ready to use.  Do you have a name for this error in mind? 

I'll also make the simple error available.

**Comment by dr-guangtou** [comment]
@alexieleauthaud 

The functions to calculate errors in `lib/compute_ds.py` (in branch `issue_29`): 

1. `get_delta_sigma_error_simple`: simple variance of weighted mean, your method. 
2. `get_delta_sigma_error_gglens`: using Melanie's (or Hironao's) method; named after the code `gglnes`.
3. `get_delta_sigma_error_jackknife`: jackknife error

**Comment by alexieleauthaud** [comment]
no, because I haven't figured out what the different assumptions are that go into the two ways of computing this, I'd be glad if someone could figure it out!
